src/effects/VST3/VST3Utils.h

```diff
@@ -52,7 +56,73 @@ class VST3Utils final
    //a human-readable form.
    static wxString MakeAutomationParameterKey(const Steinberg::Vst::ParameterInfo& info);
 
-   //Attempts to extract a unique VST3 parameter id from the key.
-   //Returns true on success.
-   static bool ParseAutomationParameterKey(const wxString& key, Steinberg::Vst::ParamID& paramId);
 };
+
+struct EffectSettings;
+
+struct VST3EffectSettings
+{
+   std::unordered_map<Steinberg::Vst::ParamID, Steinberg::Vst::ParamValue> mValues;
+};
+
+struct VST3Wrapper
+{
+   // For the time being, here we have only members that are needed
+   // to iterate parameters and extract preset state
+   //
+   Steinberg::IPtr<Steinberg::Vst::IEditController> mEditController;
+   Steinberg::IPtr<Steinberg::Vst::IComponent>      mEffectComponent;
+
+   using ParameterInfo = Steinberg::Vst::ParameterInfo;
+
+   using ParameterVisitor = std::function< bool(const ParameterInfo& pi) >;
+
+   // made mutable because it is cached once on demand
+   mutable std::vector<ParameterInfo> mParameterInfos;
+
+   // Visit parameters and keep doing so as long as visitor returns true.
+   //
+   // Return false if the parameters are not yet available
```

Comment:
For doxygen, you can write
```
/*!
   @return false if the parameters are not yet available
*/
```

There are a few alternatives of the comment syntax that doxygen also recognizes.

Reply:
I take a chance now to refresh my rusty Doxygen knowledge...